src/steps/data_preparation/utils/classification_dataset_organizer.py

```diff
@@ -0,0 +1,110 @@
+    """
+
+    def __init__(self, dataset_context: DatasetContext):
+        """
+        Initializes the organizer with a given dataset context.
+
+        Args:
+            dataset_context (DatasetContext): The dataset context to organize.
+        """
+        self.dataset_context = dataset_context
+        self.destination_dir = dataset_context.dataset_extraction_path
+
+    def organize(self):
+        """
+        Organizes the dataset by creating category directories and moving images.
+
+        Extracts category information from the COCO file, maps images to their categories,
+        and organizes the images into the respective category directories. Finally, cleans up
+        the original image directory.
+        """
+        categories = self._extract_categories()
+        image_categories = self._map_image_to_category()
+        self._organize_images(categories, image_categories)
+        self._cleanup()
+
+    def _extract_categories(self) -> Dict[int, str]:
+        """
+        Extracts the categories from the dataset's COCO file.
+
+        Returns:
+            - Dict[int, str]: A dictionary mapping category IDs to category names.
+        """
+        return {
+            category.id: category.name
+            for category in self.dataset_context.coco_file.categories
+        }
+
+    def _map_image_to_category(self) -> Dict[int, int]:
+        """
+        Maps each image to its category based on annotations in the COCO file.
+
+        Returns:
+            - Dict[int, int]: A dictionary mapping image IDs to category IDs.
+        """
+        return {
+            annotation.image_id: annotation.category_id
+            for annotation in self.dataset_context.coco_file.annotations
+        }
+
+    def _organize_images(
+        self, categories: Dict[int, str], image_categories: Dict[int, int]
+    ):
+        """
+        Creates category directories and moves images into their respective directories.
+
+        Args:
+            categories (Dict[int, str]): A mapping from category IDs to category names.
+            image_categories (Dict[int, int]): A mapping from image IDs to category IDs.
+        """
+        for image in self.dataset_context.coco_file.images:
+            image_id = image.id
+            if image_id in image_categories:
+                category_id = image_categories[image_id]
+                category_name = categories[category_id]
+                self._create_category_dir_and_copy_image(category_name, image)
+
+    def _create_category_dir_and_copy_image(self, category_name: str, image: Image):
+        """
+        Creates a directory for a category if it doesn't exist and copies an image into it.
+
+        Parameters:
+            category_name (str): The name of the category.
+            image: The image object containing the file name.
+        """
+        category_dir = os.path.join(self.destination_dir, category_name)
```

Comment:
This code can raise several exceptions (especially with `os.makedirs` and `shutil.copy`). Maybe add inside the docstring something like:

```python
"""
Raises:
        PermissionError: If there is a permission issue creating the directory or copying the file.
        FileNotFoundError: If the source image file to copy is not found.
        shutil.SameFileError: If the source and destination paths are the same.
"""
```